File: frontend/src/types/product.ts
export type Product = {
  id: number
  name: string
  price: number
  description: string
  stock_quantity: number
  created_at: string
  updated_at: string
}

export type ProductPayload = {
  name: string
  price: number
  description: string
  stock_quantity: number
}

export type ProductListResponse = {
  data: Product[]
  total: number
  page: number
  limit: number
}

export type ApiError = {
  message: string
}

File: backend/internal/domain/product.go
package domain

import (
	"context"
	"time"
)

type Product struct {
	ID            int64     `json:"id"`
	Name          string    `json:"name"`
	Price         float64   `json:"price"`
	Description   string    `json:"description"`
	StockQuantity int       `json:"stock_quantity"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type FindAllQueryParams struct {
	Page    int
	Limit   int
	Search  string
	Filters map[string]interface{}
}

type Pagination struct {
	CurrentPage int   `json:"current_page"`
	PerPage     int   `json:"per_page"`
	TotalPages  int64 `json:"total_pages"`
	TotalItems  int   `json:"total_items"`
	HasNext     bool  `json:"has_next"`
	HasPrev     bool  `json:"has_prev"`
}

type ProductRepository interface {
	// Create(product *Product) (*Product, error)
	Read(ctx context.Context, params FindAllQueryParams) ([]Product, Pagination, error)
	// ReadByID(id int64) (*Product,s
}
